concurrent/src/main/scala/scalaz/concurrent/Future.scala

```diff
@@ -302,9 +302,9 @@ object Future {
 
     // implementation runs all threads, dumping to a shared queue
     // last thread to finish invokes the callback with the results
-    override def reduceUnordered[A, M](fs: Seq[Future[A]])(implicit R: Reducer[A, M]): Future[M] =
+    override def reduceUnordered[A, M](fs: Seq[Future[A]])(implicit R: Reducer[A, M], M: Monoid[M]): Future[M] =
       fs match {
-      case Seq() => Future.now(R.zero)
+      case Seq() => Future.now(M.zero)
```

Comment:
I think not using `Seq` is a good idea. Consider either a concrete type such as `List`, or use a type-variable with `Foldable`.

Reply:
Created #1627 to address the issue separately. (as this PR did not introduced any usage of `Seq`)